service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -38,6 +40,19 @@
      * @throws IOException This exception is thrown if the object creation failed
      */
     public static ServiceClient createFromConnectionString(String connectionString, IotHubServiceClientProtocol iotHubServiceClientProtocol) throws IOException
+    {
+        return createFromConnectionString(connectionString, iotHubServiceClientProtocol, null);
+    }
+
+    /**
+     * Create ServiceClient from the specified connection string
+     * @param iotHubServiceClientProtocol  protocol to use
+     * @param connectionString The connection string for the IotHub
+     * @param proxyOptions The proxy options to use when connecting to the service. May be null if no proxy will be used.
+     * @return The created ServiceClient object
+     * @throws IOException This exception is thrown if the object creation failed
+     */
+    public static ServiceClient createFromConnectionString(String connectionString, IotHubServiceClientProtocol iotHubServiceClientProtocol, ProxyOptions proxyOptions) throws IOException
```

Comment:
What is the preferable way to set transport configurations for this SDK? Do we take in any transport settings? How do we set the keep-alive time etc?

Reply:
We currently don't have any other options beyond proxy settings. I don't see us extending other options like the C# SDK does right now. I see that as something we can address when this SDK goes track 2 instead.

Reply:
No one has asked for any settings beyond proxy support, for reference

Reply:
I see, it's better to start off with a config file, rather than placing them in the client initializer; but I agree, now that we are this close to going track 2, we should do it there directly.

Reply:
Well now I am building RegistryManagerOptions and ServiceClientOptions classes as per Priyanka, in case you are still interested in this point